### ISSUE TITLE
`Area2D` disappearing

### ISSUE THREAD

**Issue by vvh413** [grow-graphics/gd#54]
Hi.
I have encountered some strange behavior of Area2D, while trying to handle area intersection with `area_entered` signal.

Here is some code fragment with connect of `OnAreaEntered` handler to signal:
```go
func (i *InteractionManager) Ready() {
	// ...
	i.Area.AsObject().Connect(StringName.New("area_entered"), Callable.New(i.OnAreaEntered), 0)
}

func (i *InteractionManager) OnAreaEntered(o Object.Instance) {
	if a, ok := classdb.As[*InteractiveArea](o); ok {
		fmt.Println("area entered: ", a)
	}
}
```

This code results to this:

https://github.com/user-attachments/assets/30b0beaf-e10e-4600-818d-8374615cd476

So after `OnAreaEntered` handler, entered `InteractiveArea` just disappears.
`InteractiveArea` is custom `gd.Ared2D`
```go
type InteractiveArea struct {
	classdb.Extension[InteractiveArea, Area2D.Instance]
	CollisionShape CollisionShape2D.Instance
	Radius         Angle.Radians `gd:"radius"`
}
```

This might be related to #44, because if I trying to cast the entered object to `*Area2D.Instance` instead of `*InteractiveArea`, I'm getting `nil, true` as a result.
```go
func (i *InteractionManager) OnAreaEntered(o Object.Instance) {
	if a, ok := classdb.As[Area2D.Instance](o); ok {
		fmt.Println("area entered: ", a)
	}
}
```
```
area entered:  <nil>
area entered:  <nil>
```

Also this is probably related to #53, because it's actually the same project and I can't take `Area2D.Instance` as handler argument because of panic:
```
panic: reflect.Value.Convert: value of type gd.Object cannot be converted to type gd.Area2D
```
I was able to workaround this by using `Object.Instance` as handler argument or by replacing `InteractiveArea` to a manually created `Area2D` with `CollisionShape2D` in editor:
![image](https://github.com/user-attachments/assets/0e4a93e7-dcac-4c0a-8f14-8ecdf8070c0a) ![image](https://github.com/user-attachments/assets/be70c63b-c68e-4c51-a40f-bca00ca65496)
This also fixed `i.Area.GetOverlappingAreas` method. But area disappears anyway after `OnAreaEntered` handler or after taking area from `gd.ArrayOf` by index.

Initially I was trying to connect two signals `area_entered` and `area_exited` like this:
```go
i.Area.AsObject().Connect(StringName.New("area_entered"), Callable.New(i.OnAreaEntered), 0)
i.Area.AsObject().Connect(StringName.New("area_exited"), Callable.New(i.OnAreaExited), 0)
```
But this resulted in an error about trying to remove child from busy node or something. Here is the log file
[error.log](https://github.com/user-attachments/files/16443251/error.log)

**Comment by Splizard** [comment]
Thanks for reporting this, I'll have a look at this over the weekend.

**Comment by Splizard** [comment]
I notice you are trying to cast the object to a pointer type `gd.As[*gd.Area2D]`. This needs to be a value type for Godot classes. Do you still run into the problem if you use `gd.As[gd.Area2D]`?

**Comment by vvh413** [comment]
Yes, the same behavior with `gd.As[gd.Area2D]`. Just checked, this problem appears even without any casts, with just empty handler. If i create `Area2D` and connect `area_entered` signal to my handler via editor, everything works fine. So probably it is something about `Connect` in Go.

**Comment by Splizard** [comment]
Thanks for the additional information, I think this is an issue with Go Callables that accept class arguments. I suspect `gd` is incorrectly taking ownership of the arguments and freeing them when it returns.
